mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -2195,6 +2214,128 @@ impl MmCoin for TendermintCoin {
     fn on_disabled(&self) -> Result<(), AbortedError> { AbortableSystem::abort_all(&self.abortable_system) }
 
     fn on_token_deactivated(&self, _ticker: &str) {}
+
+    async fn handle_balance_stream(self, ctx: MmArc) {
+        fn generate_subscription_query(query_filter: String) -> String {
+            let q = json!({
+                "jsonrpc": "2.0",
+                "method": "subscribe",
+                "id": 0,
+                "params": {
+                    "query": query_filter
+                }
+            });
+
+            q.to_string()
+        }
+
+        let account_id = self.account_id.to_string();
+        let mut current_balances: HashMap<String, BigDecimal> = HashMap::new();
```

Comment:
`current_balances` can be part of the coin fields, when calling `my_balance`
https://github.com/KomodoPlatform/komodo-defi-framework/blob/3758cdc3e52b227db29a08ca1d18e786945fb583/mm2src/coins/tendermint/tendermint_coin.rs#L2367
 the value can be retrieved from `current_balances` if it's available to avoid additional RPC calls, I am also curious if we can use the sent/recieved tx amount to update this value straight away if it's part of the coin fields, we would have to initialize it first of course.

P.S. This is just an idea for the future.

Reply:
> if we can use the sent/recieved tx amount to update this value straight away

I don't think it's worth to deal with balance calculations depending on the spend/received amounts. When we miss an event from the socket, we will keep broadcasting the incorrect balance all the time.

> the value can be retrieved from current_balances if it's available to avoid additional RPC calls

I didn't get how `current_balances` field help us to avoid RPC calls. Can you explan a little bit?

Reply:
> I didn't get how current_balances field help us to avoid RPC calls. Can you explan a little bit?

I meant that `my_balance` is called a lot throughout the codebase, some examples:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/a2cb99bfff328da74f9fb5cdfc877ffa1ef7bafe/mm2src/mm2_main/src/lp_ordermatch.rs#L3209
https://github.com/KomodoPlatform/komodo-defi-framework/blob/a2cb99bfff328da74f9fb5cdfc877ffa1ef7bafe/mm2src/mm2_main/src/lp_ordermatch.rs#L4537
If the balance events were reliable, we could have stored the latest balance and just retrieve it in `my_balance` function if it's available. But because of this comment
> I don't think it's worth to deal with balance calculations depending on the spend/received amounts. When we miss an event from the socket, we will keep broadcasting the incorrect balance all the time.

I understand that it might not be very reliable and we should wait until this part is tested for sometime to know how much we can rely on these events for other functionalities.

Reply:
I see your point know.

Clients and the target nodes may have temporary network issues anytime, because of that reason events may be missed. I don't think we can ever rely on socket events directly.